homeassistant/components/media_player/anthemav.py

```diff
@@ -0,0 +1,179 @@
+    yield from async_add_devices([device])
+
+
+class AnthemAVR(MediaPlayerDevice):
+    """Entity reading values from Anthem AVR protocol."""
+
+    def __init__(self, hass, avr):
+        """"Initialize entity with hass, host, and port."""
+        super().__init__()
+        self.hass = hass
+        self.avr = avr
+
+    def _lookup(self, propname, dval):
+        return getattr(self.reader, propname, dval)
+
+    @property
+    def reader(self):
+        """Expose the protocol with smart wrapper."""
+        return self.avr.protocol
+
+    @property
+    def supported_media_commands(self):
+        """Return flag of media commands that are supported."""
+        return SUPPORT_ANTHEMAV
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return name of device."""
+        return self._lookup('model', DEFAULT_NAME)
+
+    @property
+    def state(self):
+        """Return state of power on/off."""
+        pwrstate = self._lookup('power', None)
+
+        if pwrstate is True:
+            return STATE_ON
+        elif pwrstate is False:
+            return STATE_OFF
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def is_volume_muted(self):
+        """Return boolean reflecting mute state on device."""
+        return self._lookup('mute', False)
+
+    @property
+    def volume_level(self):
+        """Return volume level from 0 to 1."""
+        return self._lookup('volume_as_percentage', 0.0)
+
+    @property
+    def media_title(self):
+        """Return current input name (closest we have to media title)."""
+        return self._lookup('input_name', 'No Source')
+
+    @property
+    def app_name(self):
+        """Return details about current video and audio stream."""
+        return self._lookup('video_input_resolution_text', '') + ' ' \
+            + self._lookup('audio_input_name', '')
+
+    @property
+    def source(self):
+        """Return currently selected input."""
+        return self._lookup('input_name', "Unknown")
+
+    @property
+    def source_list(self):
+        """Return all active, configured inputs."""
+        return self._lookup('input_list', ["Unknown"])
+
+    def select_source(self, source):
+        """Change AVR to the designated source (by name)."""
+        self.update_avr('input_name', source)
+        return self._lookup('input_list', ["Unknown"])
+
+    def turn_off(self):
+        """Turn AVR power off."""
+        self.update_avr('power', False)
+
+    def turn_on(self):
+        """Turn AVR power on."""
+        self.update_avr('power', True)
+
+    def set_volume_level(self, volume):
+        """Set AVR volume (0 to 1)."""
+        self.update_avr('volume_as_percentage', volume)
+
+    def mute_volume(self, mute):
+        """Engage AVR mute."""
+        self.update_avr('mute', mute)
+
+    def update_avr(self, propname, value):
```

Comment:
We should prefix this function with an underscore as well since it's also for internal use only.